lib/scnnr/connection.rb

```diff
@@ -31,22 +36,40 @@ def send_json(data)
     def send_request
       block = block_given? ? Proc.new : nil
       request = build_request(&block)
-      run_request(request)
+      with_retries do
+        Net::HTTP.start(@uri.host, @uri.port, use_ssl: use_ssl?) do |http|
+          @logger&.info("Started #{@method.upcase} #{@uri}")
+          http.request(request)
+        end
+      end
     end
 
     private
 
-    def run_request(request)
-      Net::HTTP.start(@uri.host, @uri.port, use_ssl: use_ssl?) do |http|
-        @logger&.info("Started #{@method.upcase} #{@uri}")
-        http.request(request)
+    def with_retries
+      yield
+    rescue Errno::ECONNRESET => e
```

Comment:
Cubki has a very similar feature, defined in `app/adapters/scnnr_adapter.rb#77`. Plus the error rescued there seems a bit different. Could we get rid of that old code in Cubki and handle it here? :)

Reply:
I've noticed that as well :wink: I'm all for it, but I think it's better to do it as a separate story to make sure we don't miss anything.